plugins/by-name/dap/default.nix

```diff
@@ -22,10 +22,24 @@ lib.nixvim.plugins.mkNeovimPlugin {
   deprecateExtraOptions = true;
 
   extraOptions = {
-    adapters = lib.nixvim.mkCompositeOption "Dap adapters." {
-      executables = dapHelpers.mkAdapterOption "executable" dapHelpers.executableAdapterOption;
-      servers = dapHelpers.mkAdapterOption "server" dapHelpers.serverAdapterOption;
-      pipes = dapHelpers.mkAdapterOption "pipe" dapHelpers.pipeAdapterOption;
+    adapters = lib.nixvim.mkNullOrOption' {
+      type = types.submodule {
+        freeformType = types.attrsOf types.rawLua;
+        options = {
+          executables = dapHelpers.mkAdapterOption "executable" dapHelpers.executableAdapterOption;
+          servers = dapHelpers.mkAdapterOption "server" dapHelpers.serverAdapterOption;
+          pipes = dapHelpers.mkAdapterOption "pipe" dapHelpers.pipeAdapterOption;
+        };
+      };
+      description = ''
+        Debug Adapter Protocol adapters.
+
+        You can define adapters in two ways:
+        1. Dynamic adapters using raw lua functions: `adapters.godot = { __raw = "function() ... end"; }`
+        2. Categorized structured configs: `adapters.executables.python = { command = "..."; }`
```

Comment:
Would this be better placed in an option `example`? E.g. a commented literalExpression?